enforcer-rules/README.md

```diff
@@ -3,6 +3,17 @@
 This Maven enforcer rule verifies that the transitive dependency tree of `pom.xml` does not have
 any [linkage errors](../library-best-practices/glossary.md#types-of-conflicts-and-compatibility).
 
+This rule performs the following steps:
+
+1. Builds a dependency tree of Maven artifacts with a root node having the dependencies of the
+   project
+
+   Unlike Maven's resolution algorithm, this tree includes `optional` and `provided` dependencies
```

Comment:
Updated description, starting with a normal tree.

Regarding optional and provided:

From https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management

## Optional 

> Optional dependencies - If project Y depends on project Z, the owner of project Y can mark project Z as an optional dependency, using the "optional" element. When project X depends on project Y, X will depend only on Y and not on Y's optional dependency Z. The owner of project X may then explicitly add a dependency on Z, at her option. (It may be helpful to think of **optional dependencies as "excluded by default."**)

## Provided

> This is much like compile, but indicates you expect the JDK or a container to provide the dependency at runtime. ...  This scope is only available on the compilation and test classpath, and is **not transitive**.